app/src/main/java/org/oppia/android/app/home/recentlyplayed/RecentlyPlayedFragmentPresenter.kt

```diff
@@ -282,6 +137,34 @@ class RecentlyPlayedFragmentPresenter @Inject constructor(
     }
   }
 
+  private enum class ViewType {
+    VIEW_TYPE_TITLE,
+    VIEW_TYPE_PROMOTED_STORY
+  }
+
+  private fun createRecyclerViewAdapter(): BindableAdapter<RecentlyPlayedItemViewModel> {
+    return multiTypeBuilderFactory.create<RecentlyPlayedItemViewModel, ViewType> { viewModel ->
+      when (viewModel) {
+        is PromotedStoryViewModel -> ViewType.VIEW_TYPE_PROMOTED_STORY
+        is SectionTitleViewModel -> ViewType.VIEW_TYPE_TITLE
+        else -> throw IllegalArgumentException("Encountered unexpected view model: $viewModel")
+      }
+    }
+      .registerViewDataBinder(
+        viewType = ViewType.VIEW_TYPE_TITLE,
+        inflateDataBinding = SectionTitleBinding::inflate,
+        setViewModel = SectionTitleBinding::setViewModel,
+        transformViewModel = { it as SectionTitleViewModel }
+      )
+      .registerViewDataBinder(
+        viewType = ViewType.VIEW_TYPE_PROMOTED_STORY,
+        inflateDataBinding = RecentlyPlayedStoryCardBinding::inflate,
+        setViewModel = RecentlyPlayedStoryCardBinding::setViewModel,
+        transformViewModel = { it as PromotedStoryViewModel }
+      )
+      .build()
+  }
```

Comment:
```suggestion
    }.registerViewDataBinder(
      viewType = ViewType.VIEW_TYPE_TITLE,
      inflateDataBinding = SectionTitleBinding::inflate,
      setViewModel = SectionTitleBinding::setViewModel,
      transformViewModel = { it as SectionTitleViewModel }
    ).registerViewDataBinder(
      viewType = ViewType.VIEW_TYPE_PROMOTED_STORY,
      inflateDataBinding = RecentlyPlayedStoryCardBinding::inflate,
      setViewModel = RecentlyPlayedStoryCardBinding::setViewModel,
      transformViewModel = { it as PromotedStoryViewModel }
    ).build()
  }
```

Nit: bit cleaner wrapping is possible.

Reply:
Done